### PR TITLE
Uly dataset health

### PR DESCRIPTION
**Pull request by ulya-tkch**
## Summary

> 🎯 **Purpose**:  Update dataset.health_summary() function to return same number of issues as find_label_issues() and num_label_issues(). Previously, health_summary() estimated number of issues.
>

## Impact

> 🌐 Areas Affected: dataset.py file, dataset health summary tutorial
>
> 👥 Who’s Affected: Update was added in a way to ensure backwards compatibility. So previous users calculating this using joint and without specifying labels can do so without breaking the code. Users might see updated `health_score`.

## Testing

> 🔍 Testing Done: Values match on output. Tutorial has been updated to ignore values where `pred_probs > 1.0` (aka logits were used on accident or pred_probs were normalized incorrectly)

### CODE REVIEW

**Comment by jwmueller** [comment]
closes https://github.com/cleanlab/cleanlab/issues/866

**Review by elisno — APPROVED** [review]
LGTM, nice work @ulya-tkch!
